znvis/particle/particle.py

```
@@ -113,35 +117,85 @@ def construct_mesh_list(self):
+                    elif self.director.ndim == 2:
+                        self.director = [self.director]
+        # List case
+        else:
+            n_time_steps = len(self.position)
+
+        # Check data for consistency
+        for i, position in enumerate(self.position):
+            if np.isnan(position).any():
+                raise ValueError(
+                    f"The provided position data contains NaN values at time step {i}."
+                )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)